pkg/trimaran/lowriskovercommitment/lowriskovercommitment.go

```diff
@@ -0,0 +1,258 @@
+	riskCPU := pl.computeRisk(metrics, v1.ResourceCPU, watcher.CPU, node, nodeRequestsAndLimits)
+	riskMemory := pl.computeRisk(metrics, v1.ResourceMemory, watcher.Memory, node, nodeRequestsAndLimits)
+	rank := 1 - math.Max(riskCPU, riskMemory)
+
+	klog.V(6).InfoS("Node rank", "nodeName", nodeName, "riskCPU", riskCPU, "riskMemory", riskMemory, "rank", rank)
+
+	return rank
+}
+
+// computeRisk : calculate the risk of scheduling on node for a given resource
+func (pl *LowRiskOverCommitment) computeRisk(metrics []watcher.Metric, resourceName v1.ResourceName,
+	resourceType string, node *v1.Node, nodeRequestsAndLimits *trimaran.NodeRequestsAndLimits) float64 {
+
+	var riskLimit, riskLoad, totalRisk float64
+
+	defer func() {
+		klog.V(6).InfoS("Calculated risk", "node", klog.KObj(node), "resource", resourceName,
+			"riskLimit", riskLimit, "riskLoad", riskLoad, "totalRisk", totalRisk)
+	}()
+
+	nodeRequest := nodeRequestsAndLimits.NodeRequest
+	nodeLimit := nodeRequestsAndLimits.NodeLimit
+	nodeRequestMinusPod := nodeRequestsAndLimits.NodeRequestMinusPod
+	nodeLimitMinusPod := nodeRequestsAndLimits.NodeLimitMinusPod
+	nodeCapacity := nodeRequestsAndLimits.Nodecapacity
+
+	var request, limit, capacity, requestMinusPod, limitMinusPod int64
+	if resourceName == v1.ResourceCPU {
+		request = nodeRequest.MilliCPU
+		limit = nodeLimit.MilliCPU
+		requestMinusPod = nodeRequestMinusPod.MilliCPU
+		limitMinusPod = nodeLimitMinusPod.MilliCPU
+		capacity = nodeCapacity.MilliCPU
+	} else if resourceName == v1.ResourceMemory {
+		request = nodeRequest.Memory
+		limit = nodeLimit.Memory
+		requestMinusPod = nodeRequestMinusPod.Memory
+		limitMinusPod = nodeLimitMinusPod.Memory
+		capacity = nodeCapacity.Memory
+	} else {
+		// invalid resource
+		klog.V(6).InfoS("Unexpected resource", "resourceName", resourceName)
+		return 0
+	}
+
+	// (1) riskLimit : calculate overcommit potential load
+	excess := limit - request
+	if excess < 0 {
+		excess = 0
+	}
+	maxLoad := capacity
+	if limit < capacity {
+		maxLoad = limit
+	}
+	allowed := maxLoad - request
+	if allowed < 0 {
+		allowed = 0
+	}
+	if excess > 0 && allowed < excess {
+		riskLimit = 1 - float64(allowed)/float64(excess)
+	}
+	klog.V(6).InfoS("RiskLimit", "node", klog.KObj(node), "resource", resourceName, "riskLimit", riskLimit)
+
+	// (2) riskLoad : calculate measured overcommitment
+	zeroRequest := &framework.Resource{}
+	stats, ok := trimaran.CreateResourceStats(metrics, node, zeroRequest, resourceName, resourceType)
+	if ok {
+
+		// fit a beta distribnution to the measured load stats
+		mu, sigma := trimaran.GetMuSigma(stats)
+		// adjust standard deviation due to data smoothing
+		sigma *= math.Pow(float64(pl.args.SmoothingWindowSize), 0.5)
+		// limit the standard deviation close to the allowed maximum for the beta distribution
+		sigma = math.Min(sigma, math.Sqrt(GetMaxVariance(mu)*MaxVarianceAllowance))
+
+		// calculate area under beta probability curve beyond total allocated, as overuse risk measure
```

Comment:
TBH I may not be the best guy to review the correctness/efficiency of this portion of code. Maybe @wangchen615 can give a hand on this "beta probability algorithm"?

Reply:
@wangchen615 is familiar with this plugin